src/devices/cpu/i386/i386.cpp

```diff
@@ -28,6 +28,21 @@
 #include "debug/debugcpu.h"
 #include "debug/express.h"
 
+#define LOG_MSR				(1U <<	1)
+#define LOG_INVALID_OPCODE	(1U <<	2)
+#define LOG_LIMIT_CHECK		(1U <<	3)
+#define LOG_UNEMULATED		(1U	<<	4)
+#define LOG_PM_EVENTS		(1U <<	5)
+#define LOG_PM_FAULT_GP		(1U <<  6)
+#define LOG_PM_FAULT_SS		(1U << 	7)
+#define LOG_PM_FAULT_NP		(1U <<	8)
+#define LOG_PM_FAULT_TS		(1U <<  9)
+#define LOG_PM_FAULT_DF		(1U <<	10)
+#define LOG_PM_FAULT_UD		(1U <<	11)
+
+//#define VERBOSE (LOG_PM_FAULT_GP)
+#include "logmacro.h"
```

Comment:
Should `LOG_INVALID_OPCODE` be enabled by default?  If you’re getting a lot of them, it probably means something is wrong.

Reply:
Invalid opcodes happen constantly in protected mode windows by design.  On the other hand, I think limit checks should be on by default since they should never happen.

Reply:
Does it use a reserved guaranteed-to-be-invalid opcode, or is it just using unassigned opcodes?  If x86 defines guaranteed-to-be-illegal opcodes (as SPARC and PowerPC do), we can suppress the message for those ranges.

Reply:
(Yes, I know current x86 CPUs do define such opcodes, e.g. `ud2` which is generated by clang/LLVM for what it thinks are unreachable paths.  I just don’t know how long that’s been a feature.)

Reply:
`UD2` was introduced in P6, so Pentium Pro and newer.

Reply:
They didn't.  Two examples are in enhanced mode win31 ARPL (which isn't invalid but gives a SIGILL in user mode) was used to trap into the kernel.  NTVDM uses 0xc4 0xc4 (LES EAX, ESP) for it's traps.  I'm sure there are others but those are the ones off the top of my head.